crates/holochain/src/sweettest/sweet_conductor_batch.rs

```diff
@@ -6,10 +8,24 @@ use hdk::prelude::*;
 use holochain_types::prelude::*;
 
 /// A collection of SweetConductors, with methods for operating on the entire collection
-#[derive(derive_more::From, derive_more::Into, derive_more::IntoIterator)]
+#[derive(derive_more::Into, derive_more::IntoIterator, derive_more::Deref)]
 pub struct SweetConductorBatch(Vec<SweetConductor>);
 
 impl SweetConductorBatch {
+    /// Constructor with validation
+    pub fn new(conductors: Vec<SweetConductor>) -> Self {
+        let paths: HashSet<PathBuf> = conductors
+            .iter()
+            .map(|c| c.config.environment_path.clone().into())
+            .collect();
+        assert_eq!(
```

Comment:
I've put in a similar check here https://github.com/holochain/holochain/pull/2361/files#diff-847dd7c1593e261e33e21663a9f65af856f51eb94ab563405d7e577dfa8d7c92R157 which tries to detect directory re-use across multiple tests. Given the conductors may be started in parallel it's probably worth having both checks